src/scales.js

```diff
@@ -513,21 +515,24 @@ export function scale(options = {}) {
   return scale;
 }
 
-export function exposeScales(scaleDescriptors) {
+export function exposeScales(scales) {
   return (key) => {
     if (!registry.has((key = `${key}`))) throw new Error(`unknown scale: ${key}`);
-    return key in scaleDescriptors ? exposeScale(scaleDescriptors[key]) : undefined;
+    return scales[key];
   };
 }
 
 // Note: axis- and legend-related properties (such as label, ticks and
 // tickFormat) are not included here as they do not affect the scale’s behavior.
-function exposeScale({scale, type, domain, range, interpolate, interval, transform, percent, pivot}) {
-  if (type === "identity") return {type: "identity", apply: (d) => d, invert: (d) => d};
+function exposeScale({scale, type, range, domain, interpolate, interval, transform, percent, pivot}) {
+  // The domain and range may be missing for non-position identity scales (e.g.,
+  // color), and for position identity scales, only the range is computed
+  // internally (by autoScaleRange) and then promoted to the domain here.
+  if (type === "identity") domain = range;
```

Comment:
This change is, in a sense, optional to this PR. We’ll need it when we adopt the exposed scale representation internally to render axes for position scales, but we don’t need it yet. And actually… we don’t technically need it because the axis mark could detect when the scale is an identity scale, and then use the dimensions of the chart to re-infer the domain.

I think we should add this when we need it, so we aren’t preemptively adding something we later realize we don’t need.